server/logger.js

```diff
@@ -0,0 +1,100 @@
+import winston from 'winston';
+import fs from 'fs';
+import { fileURLToPath } from 'url';
+import path from 'path';
+
+const { combine, timestamp, printf, colorize } = winston.format;
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+
+const ROOT = __dirname;
```

Comment:
In the `removeSocket` function, consider adding error handling to check if the `usertoken` exists in `socket_user_map` before attempting to delete it. This will prevent potential errors when trying to delete a non-existent entry.
```diff
+  if (usertoken in socket_user_map) {
+    delete socket_user_map[usertoken];
+  }
```


    *This comment was auto-generated by **Code Bot** to provide suggestions and improvements.*